python-lib/rest_api_recipe_session.py

```diff
@@ -117,7 +118,13 @@ def retrieve_next_page(self, is_raw_output):
                         base_row.update(parse_keys_for_json(row))
                         base_row.update(self.initial_parameter_columns)
                         page_rows.append(base_row)
-
+                else:
+                    json_response = decode_csv_data(json_response)
```

Comment:
same remark as for dataset#generate_rows logic; csv decoding looks like a fallback when nothing works. I'm wondering if we can handle this format prior to that, testing response header content-type as for xml.

Reply:
The recipe's behavior is modeled on the connector's.